cmd/revad/runtime/runtime.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2018-2019 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"path"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"contrib.go.opencensus.io/exporter/jaeger"
+	"github.com/cs3org/reva/cmd/revad/internal/config"
+	"github.com/cs3org/reva/cmd/revad/internal/grace"
+	"github.com/cs3org/reva/pkg/logger"
+	"github.com/cs3org/reva/pkg/rgrpc"
+	"github.com/cs3org/reva/pkg/rhttp"
+	"github.com/gofrs/uuid"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"go.opencensus.io/plugin/ocgrpc"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+// Run runs a reva server with the given config file and pid file.
+func Run(config string, pidFile string) {
```

Comment:
Can we pass in the map[string]interface{} here so we don't have to read the config from disk? An io.Reader would work as well, but passing in a map would be the least overhead.

Reply:
@butonic of course :)